### PR TITLE
Support configurable AggregationTemporality in exporters; add OTLP missing sum point temporality/monotonic fields

### LINKED ISSUE #1288: Exported OTLP metric Int/DoubleSum data points lack temporality / is_monotonic fields
The current OTLP exporter does not set two fields in the IntSum and DoubleSum data points. The testing framework is also not validating these fields.

e.g., in `otlp_metric_test.go` the function `runMetricExportTest` does not check the expected temporality or monotonicity. There is no mention of these fields in the `internal/transform` directory.

### PR DESCRIPTION
**Pull request by jmacd**
The metric export API has a type `ExportKind` that describes which aggregation temporality to use, Delta or Cumulative. There was a special value, `PassThroughExporter`, indicating the intent to use the natural aggregation temporality for sums, since that leads to a stateless configuration. The associated type `ExportKindSelector` is used to choose an export kind, and there was confusion over actual export kinds (Delta or Cumualtive) and selectors, which are policies that decide the export kind.

This PR removes `PassThroughExporter`, replacing it with `StatelessExportKindSelector()`. This makes it so that `ExportKind` does not implement `ExportKindSelector`, which was being used for constant ExportKindSelector policies (and made the code difficult to read). Constant policies are now available as `DeltaExportKindSelector()` and `CumulativeExportKindSelector()`.

Updates the OTLP exporter to use `CumulativeExportKindSelector()` by default, as stated in https://github.com/open-telemetry/opentelemetry-specification/issues/731. Adds `WithExportKindSelector()` option to the OTLP exporter to configure alternate policies. Using `WithExportKindSelector(StatelessExportKindSelector())` results in the former `PassThroughExporter` behavior.

Add AggregationTemporality and IsMonotonic fields in OTLP exporter. Fixes #1288.

### CODE REVIEW

**Review by codeboten — APPROVED** [review]
👍

**Comment by jmacd** [comment]
Someone privately asked if this PR should be split into smaller parts, as there are several things happening. I see these as all part of one bug fix, so I'm inclined to leave it the way it is, but just for the sake of discussion:

This could be sequenced as: (1) remove the PassThrough ExportKind, (2) change OTLP exporter to pass ExportKindSelector to where it's needed, (3) add the missing fields.

Note that (1) and (2) are independent, and that (3) depends on both. However, this assumes an approach that is not the only way forward. Instead of passing the ExportKindSelector into the Exporter where it re-calculates the export record's temporality (which requires it to be consistent), we could just add a field in the export record to say whether the record with its two timestamps is notionally DELTA or CUMULATIVE, for that's what we lack. We know the timestamps of the interval but we don't know whether to call the interval a DELTA or a CUMULATIVE.

I have another way to reason about this change of behavior. The last release uses OTLP v0.3, which lacked temporality fields. Although the existing code says that "passthrough" is the default export kind for OTLP, there were no fields in OTLP v0.3 to reflect temporality. We've renamed that concept "stateless" in this PR, and it's a critical concept IMO, but for months we've stated that the default would change to CUMULATIVE for OTLP and we couldn't do that until v0.5 support, so I see this "fix" as just part of finishing v0.5 support, not some elaborate new behavior about configurable aggregation temporality. (Aggregation temporality was always configurable, just not in a way the exporter can see.)

Therefore, @open-telemetry/go-approvers I'd like to review this as-is in the interest of a speedy release.

File: exporters/otlp/otlp_metric_test.go [hunk 1]
```diff
@@ -695,13 +700,78 @@ func TestResourceInstLibMetricGroupingExport(t *testing.T) {
 	)
 }
 
+func TestStatelessExportKind(t *testing.T) {
+	type kase struct {
```
Comment (krnowak) [inline exporters/otlp/otlp_metric_test.go:704]: That reads like german Käse (which means cheese). `testcase` maybe?

**Review by krnowak — APPROVED** [review]
Looks good.

File: exporters/otlp/otlp_metric_test.go [hunk 2]
```diff
@@ -695,13 +700,78 @@ func TestResourceInstLibMetricGroupingExport(t *testing.T) {
...
+		name  string
+		ikind otel.InstrumentKind
+		tempo metricpb.AggregationTemporality
+		mono  bool
```
Comment (MrAlias) [inline exporters/otlp/otlp_metric_test.go:708]: Might be useful to use full words for these so new users are not confused.

**Review by MrAlias — APPROVED** [review]

File: exporters/otlp/otlp_metric_test.go (hunk 2 above)
Comment (jmacd) [reply exporters/otlp/otlp_metric_test.go:708]: Done.

File: exporters/otlp/otlp_metric_test.go (hunk 1 above)
Comment (jmacd) [reply exporters/otlp/otlp_metric_test.go]: Done.

**Comment by jmacd** [comment]
I think this is ready to merge.

**Review by XSAM — APPROVED** [review]

**Pull request merged**